lib/ansible/modules/cloud/vmware/vmware_spbm.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Prasanna Nanda <pnanda@cloudsimple.com>
+
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: vmware_spbm
+short_description: VMware Storage Profile Utilities
+description:
+    - Create, Destroy VM Storage Profiles
+version_added: 2.4
+author:
+    - Prasanna Nanda <pnanda@cloudsimple.com>
+notes:
+    - Tested on vSphere 6.5
+requirements:
+    - "python >= 2.7"
+    - PyVmomi
+options:
+    profile_name:
+        description:
+            - Name of the storage profile to Create
+        required: True
+    state:
+        description:
+            - Valid values are present or absent
+        required: True
+    description:
+        description:
+            - Some Text Describing the Storage profile
+        required: False
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = r'''
+  - name: Create a Storage Profile
+    vmware_spbm:
+      hostname: A vCenter host
+      username: vCenter username
+      password: vCenter password
+      profile_name: Name of the storage profile to create
+      description: Text Describing Storage Profile
+      state: present|absent
+'''
+
+RETURN = ''' # '''
+
+try:
+    from pyVmomi import vim, pbm, vmodl
+    from pyVim.connect import SoapStubAdapter
+    from pyVim.connect import SmartConnect, Disconnect
+    HAS_PYVMOMI = True
+except ImportError:
+    HAS_PYVMOMI = False
+
+from ansible.module_utils.vmware import vmware_argument_spec, connect_to_api
+import ssl
+import atexit
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.spbm_client import SPBMClient
+
+
+class vmware_spbm(object):
+    def __init__(self, module):
+        self.module = module
+        self.name = module.params['profile_name']
+        self.description = module.params['description']
+        self.desiredstate = module.params['state']
+        self.rules = module.params['rules']
+        self.currentstate = 'absent'
+        try:
+            context = None
```

Comment:
Why I could not use connect_to_api of vmware.py, because SPBM needs to
have access to ServiceInstance for Stub Generation. connect_to_api returns
a ServiceContent